### PR TITLE
[inetstack] Fixing issue #244: Adding remove functionality to remove an entry from the HashTTLCache

### PR DESCRIPTION
**Pull request by iyzhang**
This is one of our oldest issues, so I thought I would fix it just to go through the coding guidelines. I added a new unit test for the functionality.

### CODE REVIEW

**Review by anandbonde — APPROVED** [review]
Looks good. Do we have an issue to track making use of the "remove" functionality? Now that we have a working remove, we should use it at appropriate the place(s).

**Review by ppenna — CHANGES_REQUESTED** [review]
In addition, please rewrite the commit history and stick to the conventions.

```
git branch -m enhancement-inetstack-ttl-chache
git rebase -i dev
# Choose 'reword'
```

File: src/rust/inetstack/collections/hashttlcache/mod.rs [hunk 1]
```diff
@@ -115,7 +115,8 @@ where
 
     /// Removes an entry from the cache.
     pub fn remove(&mut self, _key: &K) -> Option<V> {
-        todo!()
+        let (k, v) = self.map.remove_entry(_key).unwrap();
```
[the remark this answers is not in the document]
Comment (iyzhang) [reply src/rust/inetstack/collections/hashttlcache/mod.rs:118]: I assumed that remove was suppose to return the removed element.

[the remark this answers is not in the document]
Comment (ppenna) [reply src/rust/inetstack/collections/hashttlcache/mod.rs:118]: It does, but what if the requested key is NOT in the map? Then `unwrap()` will cause the system to panic. Better than this, is to fail and return `None`.

**Pull request closed without merging**